src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/prisonersearch/PrisonerSearchService.kt

```diff
@@ -1,39 +1,64 @@
 package uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.prisonersearch
 
 import mu.KotlinLogging
-import org.springframework.http.HttpStatus
 import org.springframework.stereotype.Service
-import org.springframework.web.reactive.function.client.WebClientResponseException
-import reactor.core.publisher.Mono
+import uk.gov.justice.digital.hmpps.prisonersearch.model.Prisoner
+import uk.gov.justice.digital.hmpps.prisonersearch.model.PrisonerMatch
+import uk.gov.justice.digital.hmpps.prisonersearch.model.PrisonerMatches
+import uk.gov.justice.digital.hmpps.prisonersearch.model.PrisonerMatches.MatchedBy.aLLSUPPLIED
+import uk.gov.justice.digital.hmpps.prisonersearch.model.PrisonerMatches.MatchedBy.aLLSUPPLIEDALIAS
 import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.barcode.BarcodeRecipient
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.security.UserContext
 
 private val log = KotlinLogging.logger {}
 
 @Service
-class PrisonerSearchService(private val prisonerSearchClient: PrisonerSearchClient) {
+class PrisonerSearchService(private val prisonerSearchClient: PrisonerSearchClient, private val userContext: UserContext) {
   fun lookupPrisoner(barcodeRecipient: BarcodeRecipient) {
     val prisonerSearchRequest = PrisonerSearchRequest(barcodeRecipient)
     prisonerSearchClient.matchPrisoners(prisonerSearchRequest)
-      .onErrorResume {
-        with(it) {
-          if (this is WebClientResponseException) {
-            if (statusCode == HttpStatus.FORBIDDEN) {
-              log.info { "Client token does not have correct role to call prisoner-search" }
-            } else {
-              log.error { "Failed to call prisoner-search - statusCode $statusCode" }
-            }
-          } else {
-            log.error("Failed to call prisoner-search", it)
-          }
+      .subscribe { processPrisonerMatches(prisonerSearchRequest, it) }
+  }
+
+  private fun processPrisonerMatches(prisonerSearchRequest: PrisonerSearchRequest, prisonerMatches: PrisonerMatches) {
+    with(prisonerMatches) {
```

Comment:
This code to process the prisoner matches is a bit pedestrian and a bit ugly (nested if's etc 😢 ), but until we start seeing and understanding what the data looks like it felt sensible to do this 'long hand' this way

Reply:
I wasn't too worried about a matching algorithm yet - I thought this phase is just about gathering data to see the search results fall out. But ok for a first shout - in the longer term I suspect you will have a matching service with a gazillion tests against it.